RO/collectives: add linear algorithms using RPut/Rget

edgargabriel opened the pull request:
- make broadcast, alltoall and fcollect use a simple linear algorithm using MPI_RPut/Rget, but without blocking in the execution. This fixes a hang and blocking observed with previous implementations.
- remove the to_all interfaces, since they have been deprecated.
- remove the active-set interfaces, since they have been removed from rocSHMEM
- disable allocating the ata_buffer on the RO teams structure, since it is not used at the moment and saves as 128MB per team

edgargabriel committed:
RO/collectives: add linear algorithms using RPut/Rget

- make broadcast, alltoall and fcollect use a simple linear algorithm
  using MPI_RPut/Rget, but without blocking in the execution
- remove the to_all interfaces, since they have been deprecated.
- remove the active-set interfaces, since they have been removed from
  rocSHMEM

avinashkethineedi commented on src/reverse_offload/mpi_transport.cpp:
```diff
@@ -469,568 +357,134 @@ void MPITransport::team_broadcast(void *dst, void *src, int size, int win_id,
-  barrier(contextId, status, false, comm_cluster);
-  barrier(contextId, status, blocking, comm_ring);
+  NET_CHECK(MPI_Win_flush_all(bp->heap_window_info[win_id]->get_win()));
+  barrier(contextId, status, blocking, comm);
+  quiet(contextId,status);
```
```suggestion
  barrier(contextId, nullptr, false, comm);
  quiet(contextId, status);
```
to avoid notifying the GPU thread when barrier completes

edgargabriel committed:
avoid notification after barrier

Co-authored-by: Avinash Kethineedi <avinash.kethineedi@amd.com>

edgargabriel committed:
disable allocation of ata_buffer

a temporary buffer of 128MB was allocated when creating a team. In
previous versions of the code, that buffer was used by some collective
operations. This is not the case for now. Therefore, do not allocate the
buffer for now. I am not removing the element itself from teh
structure, since we might need it in future versions again.
